### PR TITLE
Revise building functionality logic #143

### PR DESCRIPTION
**Pull request by longshuicy**
test by comment out any of the below in the `test_buildingfunctionality.py`
``` 
 bldg_func.load_remote_input_dataset("substations_damage_mcs_samples", "5f0f71bab922f96f4e9a7511")
 bldg_func.load_remote_input_dataset("poles_damage_mcs_samples", "5f0f7231b922f96f4e9a7538")
 bldg_func.load_remote_input_dataset("interdependency_dictionary", "5f0f7311feef2d758c47cfab")
```

### CODE REVIEW

**Review by nunezco2 — APPROVED** [review]

File: pyincore/analyses/buildingfunctionality/buildingfunctionality.py [hunk 1]
```diff
@@ -86,13 +86,34 @@ def get_spec(self):
...
+        poles_dataset = self.get_input_dataset("poles_damage_mcs_samples")
+        if poles_dataset is not None:
+            poles_df = poles_dataset.get_dataframe_from_csv().set_index("guid")
+        else:
+            poles_df = None
+
+        # All three above needs to be present at the same time or none
+        if not all(epf is None for epf in [interdependency_dict, substations_df, poles_df]) \
+                and not all(epf is not None for epf in [interdependency_dict, substations_df, poles_df]):
+            raise ValueError("If you need to consider electric power availability in the analysis, please provide "
```
Comment (navarroc) [inline pyincore/analyses/buildingfunctionality/buildingfunctionality.py:114]: I suggest a slight rewording:
To consider electric power availability in the analysis, please provide pole damage samples, substation damage samples, and interdependency.

**Review by navarroc — APPROVED** [review]

**Comment by ywkim312** [comment]
maybe update changelog?

**Comment by navarroc** [comment]
> maybe update changelog?

I think we're doing that in the release branch, right? Otherwise we end up with conflicts for every merge that modifies the changelog.

**Comment by ywkim312** [comment]
> > maybe update changelog?
> 
> I think we're doing that in the release branch, right? Otherwise we end up with conflicts for every merge that modifies the changelog.

yes, you're right. I sent a message to Chen about ignore my comment regarding update changlog, but forgot that I commented on your PR as well. Sorry about that.

**Pull request merged**